src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/InterpreterLanguagesDynamicListUpdater.java

```diff
@@ -73,29 +80,68 @@ public PreSubmitCallbackResponse<AsylumCase> handle(
-            .orElse(false);
-        boolean hasSignLang = callback.getCaseDetailsBefore()
-            .map(caseDetails -> caseDetails.getCaseData().read(APPELLANT_INTERPRETER_SIGN_LANGUAGE, InterpreterLanguageRefData.class).isPresent())
-            .orElse(false);
+        AsylumCase asylumCaseBefore = callback.getCaseDetailsBefore()
+            .map(CaseDetails::getCaseData)
+            .orElse(asylumCase);
 
-        if (Objects.equals(pageId, DRAFT_HEARING_REQUIREMENTS_PAGE_ID)) {
-            if (!hasSpokenLang) {
-                populateDynamicList(asylumCase, INTERPRETER_LANGUAGES, APPELLANT_INTERPRETER_SPOKEN_LANGUAGE);
+        String pageId = callback.getPageId();
+        Event event = callback.getEvent();
+
+        Optional<CaseDetails<AsylumCase>> optionalCaseDetailsBefore = callback.getCaseDetailsBefore();
+
+        Optional<InterpreterLanguageRefData> appellantSpokenLanguage = optionalCaseDetailsBefore
+            .flatMap(caseDetailsBefore ->
+                caseDetailsBefore.getCaseData()
+                    .read(APPELLANT_INTERPRETER_SPOKEN_LANGUAGE, InterpreterLanguageRefData.class));
+
+        Optional<InterpreterLanguageRefData> appellantSignLanguage = optionalCaseDetailsBefore
+            .flatMap(caseDetailsBefore ->
+                caseDetailsBefore.getCaseData()
+                    .read(APPELLANT_INTERPRETER_SIGN_LANGUAGE, InterpreterLanguageRefData.class));
+
+        boolean shouldPopulateSpokenLanguage = appellantSpokenLanguage
+            .map(appellantSpoken -> appellantSpoken.getLanguageRefData() == null)
+            .orElse(true);
+
+        boolean shouldPopulateSignLanguage = appellantSignLanguage
+            .map(appellantSign -> appellantSign.getLanguageRefData() == null)
+            .orElse(true);
+
+        // POPULATE APPELLANT INTERPRETER LANGUAGE DYNAMIC LIST IF NECESSARY
+        // TO BE RUN FOR: 1) DRAFT_HEARING_REQUIREMENTS 2) UPDATE_HEARING_REQUIREMENTS
+
+        //boolean hasRefDataSpokenLangPopulated = callback.getCaseDetailsBefore()
+        //    .map(caseDetails -> caseDetails.getCaseData().read(APPELLANT_INTERPRETER_SPOKEN_LANGUAGE, InterpreterLanguageRefData.class)
+        //        .orElse(new InterpreterLanguageRefData(new DynamicList(null, Collections.emptyList()), Collections.emptyList(), null))
+        //        .getLanguageRefData() != null)
+        //    .orElse(false);
+        //
+        //
+        //boolean hasRefDataSignLangPopulated = callback.getCaseDetailsBefore()
+        //    .map(caseDetails -> caseDetails.getCaseData().read(APPELLANT_INTERPRETER_SIGN_LANGUAGE, InterpreterLanguageRefData.class)
+        //        .orElse(new InterpreterLanguageRefData(new DynamicList(null, Collections.emptyList()), Collections.emptyList(), null))
+        //        .getLanguageRefData() != null)
+        //    .orElse(false);
```

Comment:
If this piece is not needed why not delete?

Reply:
This was dead code I forgot to delete. I deleted it now and left the comment.